src/module.c

```diff
@@ -10987,10 +11123,429 @@ sds genModulesInfoString(sds info) {
+        return REDISMODULE_ERR;
+    }
+    return REDISMODULE_OK;
+}
+
+/* This is a series of set functions for each type that act as dispatchers for 
+ * config.c to call module set callbacks. */
+int setModuleBoolConfig(ModuleConfig *config, int val, const char **err) {
+    int return_code = config->set_fn.set_bool(config->name, val, config->privdata, err);
+    return return_code == REDISMODULE_OK ? 1 : 0;
+}
+
+int setModuleStringConfig(ModuleConfig *config, sds strval, const char **err) {
+    if (!strval) return 1;
+    RedisModuleString *new = createStringObject(strval, sdslen(strval));
+    int return_code = config->set_fn.set_string(config->name, new, config->privdata, err);
+    decrRefCount(new);
+    return return_code == REDISMODULE_OK ? 1 : 0;
+}
+
+int setModuleEnumConfig(ModuleConfig *config, int val, const char **err) {
+    int return_code = config->set_fn.set_enum(config->name, val, config->privdata, err);
+    return return_code == REDISMODULE_OK ? 1 : 0;
+}
+
+int setModuleNumericConfig(ModuleConfig *config, long long val, const char **err) {
+    int return_code = config->set_fn.set_numeric(config->name, val, config->privdata, err);
+    return return_code == REDISMODULE_OK ? 1 : 0;
+}
+
+/* This is a series of get functions for each type that act as dispatchers for 
+ * config.c to call module set callbacks. */
+int getModuleBoolConfig(ModuleConfig *module_config) {
+    return module_config->get_fn.get_bool(module_config->name, module_config->privdata);
+}
+
+sds getModuleStringConfig(ModuleConfig *module_config) {
+    RedisModuleString *val = module_config->get_fn.get_string(module_config->name, module_config->privdata);
+    return val ? sdsdup(val->ptr) : NULL;
+}
+
+int getModuleEnumConfig(ModuleConfig *module_config) {
+    return module_config->get_fn.get_enum(module_config->name, module_config->privdata);
+}
+
+long long getModuleNumericConfig(ModuleConfig *module_config) {
+    return module_config->get_fn.get_numeric(module_config->name, module_config->privdata);
+}
+
+/* This function takes a module and a list of configs stored as sds NAME VALUE pairs.
+ * It attempts to call set on each of these configs. */
+int loadModuleConfigs(RedisModule *module) {
+    listIter li;
+    listIter li_supplied;
+    listNode *ln;
+    listNode *ln_supplied;
+    int count = 0;
+    sds *config_pair;
+    const char *err = NULL;
+
+    listRewind(module->module_configs, &li);
+    while ((ln = listNext(&li))) {
+        ModuleConfig *module_config = ln->value;
+        int config_supplied = 0;
+        sds config_name = sdscatfmt(sdsempty(), "%s.%s", module->name, module_config->name);
+        listRewind(server.module_configs_queue, &li_supplied);
+        while ((ln_supplied = listNext(&li_supplied))) {
+            sds supplied_config = ln_supplied->value;
+            config_pair = sdssplitargs(supplied_config, &count);
+            if (count != 2) {
```

Comment:
Made a related comment, but I think we should make sure this is can't happen so we can serverPanic here instead.